usersession/userd/privileged_desktop_launcher.go

```diff
@@ -118,6 +129,20 @@ func (s *PrivilegedDesktopLauncher) OpenDesktopEntry(desktopFileID string, sende
 	return nil
 }
 
+// argumentsSecurityCheck ensures that none of the arguments passed
+// by begins with "-", to avoid passing parameters to the application.
+func argumentsSecurityCheck(arguments []string) error {
+	if arguments == nil {
+		return nil
+	}
+	for i := 0; i < len(arguments); i++ {
```

Comment:
You could replace this with:

    for i := range arguments {

Or even

    for _, arg := range arguments {

Reply:
Done.